tests/linalg_test.py

```diff
@@ -2357,5 +2358,59 @@ def testPascal(self, n, kind):
     self._CompileAndCheck(jsp_fun, args_maker)
 
 
+class TestSolveSylvester(jtu.JaxTestCase):
+  # float32 is not close enough for assert
+  jax.config.update("jax_enable_x64", True)  # Ensure float64 for higher precision
+
+  @jtu.sample_product(
+    shape=[(2, 3), (4, 6), (5, 7), (100, 300)],
+  )
+  def test_solve_sylvester(self, shape):
+    n, m = shape
+
+    A = random.normal(random.key(0), shape=(n, n))
+    B = random.normal(random.key(1), shape=(m, m))
+    X_true = random.normal(random.key(2), shape=(n, m))
+
+    C = A @ X_true + X_true @ B
+    with jax.numpy_dtype_promotion('standard'):
```

Comment:
If all three inputs are the same dtype, then the function should execute without error under strict dtype promotion. If it does not, it should be fixed within the `sylvester` implementation: it means you're generating intermediate arrays of mixed type, and instead you should explicitly cast them within array operations.